config/src/main/java/org/springframework/security/config/annotation/web/configuration/OAuth2ClientConfiguration.java

```diff
@@ -84,5 +90,11 @@ public void setAuthorizedClientRepository(List<OAuth2AuthorizedClientRepository>
 				this.authorizedClientRepository = authorizedClientRepositories.get(0);
 			}
 		}
+
+		@Autowired
+		public void setAccessTokenResponseClient(
+				Optional<OAuth2AccessTokenResponseClient<OAuth2ClientCredentialsGrantRequest>> accessTokenResponseClient) {
```

Comment:
If the user registers more than one `@Bean` of `OAuth2AccessTokenResponseClient<OAuth2ClientCredentialsGrantRequest>` that this will error. This case is already handled by `setClientRegistrationRepository` and `setAuthorizedClientRepository`. We should also have a test for this case.